tensorflow/python/kernel_tests/one_hot_op_test.py

```diff
@@ -45,62 +45,74 @@ def _testBothOneHot(self, truth, expected_err_re=None, raises=None, **inputs):
     self._testOneHot(truth, False, expected_err_re, raises, **inputs)
 
   def _testBasic(self, dtype):
-    indices = np.asarray([0, 2, -1, 1], dtype=np.int64)
```

Comment:
i dont think you have to modify all these tests.  just add one new test where the indices dtype is an int32.  (say, take the most complicated of the existing tests and copy it in a new method)